samples/snippets/core/system-text-json/csharp/Utf8ReaderPartialRead.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+
+namespace SystemTextJsonSamples
+{
+    public class Utf8ReaderPartialRead
+    {
+        public static void Run()
+        {
+            var jsonString = @"{
+                ""Date"": ""2019-08-01T00:00:00-07:00"",
+                ""Temperature"": 25,
+                ""TemperatureRanges"": {
+                    ""Cold"": { ""High"": 20, ""Low"": -10 },
+                    ""Hot"": { ""High"": 60, ""Low"": 20 }
+                },
+                ""Summary"": ""Hot"",
+            }";
+
+            byte[] bytes = Encoding.UTF8.GetBytes(jsonString);
+            var stream = new MemoryStream(bytes);
+
+            var buffer = new byte[4096];
+
+            // Fill the buffer.
+            // For this snippet, we're assuming the stream is open and has data.
+            // If it might be closed or empty, check if the return value is 0.
+            stream.Read(buffer);
+
+            var reader = new Utf8JsonReader(buffer, isFinalBlock: false, state: default);
+            Console.WriteLine($"String in buffer is: {Encoding.UTF8.GetString(buffer)}");
+
+            // Search for "Summary" property name
+            while (reader.TokenType != JsonTokenType.PropertyName || !reader.ValueTextEquals("Summary"))
+            {
+                if (!reader.Read())
+                {
+                    // Not enough of the JSON is in the buffer to complete a read.
+                    GetMoreBytesFromStream(stream, ref buffer, ref reader);
+                }
+            }
+
+            // Found the "Summary" property name.
+            Console.WriteLine($"String in buffer is: {Encoding.UTF8.GetString(buffer)}");
```

Comment:
What does  this output? Do we care and if so, should we show the reader in the comment as well of what this would output? The same applies to all other `Console.WriteLine` calls.

Reply:
What it shows depends on the buffer size -- at 4096, which the snippet has now, the entire sample JSON is displayed.  If you change it to 10, it shows 10 byte sections of the JSON at a time until the buffer is increased to 20 and then to 40, after which the sections shown are correspondingly bigger. It basically just gives a visual record of when and why the buffer is increased.